services/QuillLMS/app/lib/long_process_notifier.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class LongProcessNotifier < ApplicationService
+  attr_reader :error, :threshold, :options
+
+  def initialize(error, threshold, options = {})
+    @error = error
+    @threshold = threshold
+    @options = options
+  end
+
+  def run
+    start = self.class.current_time
+    result = yield
+    runtime = self.class.current_time - start
+    return result if runtime < @threshold.to_i
```

Comment:
Take or leave, but since result is always returned, this might be more readable
```ruby
if runtime < @threshold.to_i
  ErrorNotifier.report(@error, @options.merge({time_to_execute: runtime}))
end

result